config/initializers/attribute_types.rb

```diff
@@ -1,9 +1,9 @@
 class Gbp < ActiveModel::Type::Decimal
   def cast(value)
-    if value.is_a?(Numeric) || value.blank? || value.gsub(/[0-9,.£]/, "").empty?
+    if value.is_a?(Numeric) || value.blank? || value.gsub(/[0-9,.£]+\s?\z/, "").empty?
       # When a monetary value is provided by a form submission, we are only interested
       # in the value to 2 decimal places, and discard any more fine detail immediately
-      super(value&.to_s&.delete(",")&.delete("£"))&.round(2)
+      super(value&.to_s&.chomp&.delete(",")&.delete("£"))&.round(2)
```

Comment:
`chomp` doesn't remove trailing spaces, it removes newlines. I think you're thinking of `strip`?

Reply:
But also, you don't actually need to make this change at all, because whitespace gets stripped by the superclass `cast` method anyway.

Reply:
Oh so the thing that needs to change is making something like "123 " reach this line at all, which it currently isn't because of the regex in the `if` statement?

Reply:
Exactly. The assumption was that the 'if' statement would only evaluate `false` in situations where validation also failed, so that we could show a validation message and prepopulate the form with the invalid value provided by the user exactly as entered. The bug arises where the if statement is falsey but the validation _passes_, leading to the value exactly as entered being stored in the session and consequently being passed on to CFE.